qiskit/primitives/primitive_job.py

```diff
@@ -40,9 +40,8 @@ def submit(self):
         if self._future is not None:
             raise JobError("Primitive job has already been submitted.")
 
-        with ThreadPoolExecutor(max_workers=1) as executor:
-            future = executor.submit(self._function, *self._args, **self._kwargs)
-        self._future = future
+        executor = ThreadPoolExecutor(max_workers=1)  # pylint: disable=consider-using-with
+        self._future = executor.submit(self._function, *self._args, **self._kwargs)
```

Comment:
Who is responsible for shutting down the executor after this change?

Reply:
Yeah, I have the same thing when I implemented the original version.
But I found `AerJob` uses this technique. https://github.com/Qiskit/qiskit-aer/blob/13937fdb596142006bf00caf5676da13b43dfb5a/qiskit_aer/jobs/aerjob.py#L58

Reply:
I didn't know `shutdown(wait=False)`, but it certainly sounds better.

Reply:
Aer's setup is a bit weird, I think - when that code line was originally written (4 years ago), Aer was using a global executor that lived for the entire process.  `shutdown` doesn't really do much with `wait=False` other than mark the executor as no longer usable, and since it's no longer accessible after this function ends, I don't think it's required to do.

(caveat: I do very little async programming, so I'm not fully up-to-date on best practices)

Reply:
There is a limit of number of threads depending on OS, I guess. The following code fails to make a `ThreadPoolExecutor` object as follows. We may need to take care of the error.

```python
from concurrent.futures import ThreadPoolExecutor
from time import sleep


def func():
    sleep(2)


for i in range(10000):
    try:
        e = ThreadPoolExecutor(max_workers=1)
        e.submit(func)
    except RuntimeError as ex:
        print(ex)
        print(i)
        break
```

output with Python 3.10 on macOS 12.6.1
```
can't start new thread
4095
```

Reply:
I doubt anyone would use it in such a strange way, and at least we haven't had any problems with Aer Backend over the years, so I believe it's fine.

Do you have the solution to the problem?

Reply:
I have no idea. If it works fine with Aer for years, I think it's no problem.